sync/src/main/java/io/apicurio/sync/RegistryClientProducer.java

```diff
@@ -23,7 +30,15 @@ public class RegistryClientProducer {
 
     @Produces
     public RegistryClient registryClient() {
-        return RegistryClientFactory.create(registryUrl);
+        final String tokenEndpoint = System.getenv("AUTH_TOKEN_ENDPOINT");
+        if (tokenEndpoint != null) {
+            final String authClient = System.getenv("AUTH_CLIENT_ID");
+            final String authSecret = System.getenv("AUTH_CLIENT_SECRET");
+            Auth auth = new OidcAuth(new JdkHttpClient(tokenEndpoint, Collections.emptyMap(), null, new AuthErrorHandler()), authClient, authSecret);
+            return RegistryClientFactory.create(new JdkHttpClient(registryUrl, Collections.emptyMap(), auth, new AuthErrorHandler()));
```

Comment:
After deep diving,  I found that this handler should be `AuthErrorHandler` to catch `ArtifactAlreadyExistsException` (e.g. https://github.com/Apicurio/apicurio-registry-content-sync-operator/blob/579db7e391088a3084367d14c2e4e14de344ceb0/sync/src/main/java/io/apicurio/sync/controller/ArtifactController.java#L260)


```suggestion
            Auth auth = new OidcAuth(new JdkHttpClient(tokenEndpoint, Collections.emptyMap(), null, new io.apicurio.registry.rest.client.impl.ErrorHandler()), authClient, authSecret);
            return RegistryClientFactory.create(new JdkHttpClient(registryUrl, Collections.emptyMap(), auth, new io.apicurio.registry.rest.client.impl.ErrorHandler()));
```

I created a PR for building the image that supports OAuth and confirmed that it worked as expected.
https://github.com/Apicurio/apicurio-registry-content-sync-operator/pull/144
(Though, it has some issues in testing and generating beans.)